test/e2e/network/service.go

```diff
@@ -3283,42 +3282,74 @@ var _ = SIGDescribe("ESIPP [Slow]", func() {
 			framework.Failf("Service HealthCheck NodePort still present")
 		}
 
-		endpointNodeMap, err := jig.GetEndpointNodes()
+		config := e2enetwork.NewNetworkingTestConfig(f, false, false)
+
+		epNodes, err := jig.ListNodesWithEndpoint()
 		framework.ExpectNoError(err)
-		noEndpointNodeMap := map[string][]string{}
+		// map from name of nodes with endpoint to internal ip
+		endpointNodeMap := make(map[string]string)
+		// map from name of nodes without endpoint to internal ip
+		noEndpointNodeMap := make(map[string]string)
+		for _, node := range epNodes {
+			ips := e2enode.GetAddresses(&node, v1.NodeInternalIP)
+			if len(ips) < 1 {
+				framework.Failf("No internal ip found for node %s", node.Name)
+			}
+			endpointNodeMap[node.Name] = ips[0]
+		}
+		// it is assumed that there is only a single node with the endpoint
```

Comment:
Move this comment to line 3290, i.e.

// map from name of nodes with endpoint to internal ip
// it is assumed that there is only a single node with the endpoint
endpointNodeMap := make(map[string]string)

Reply:
please check the latest commit